server/player/player.go

```diff
@@ -584,9 +584,24 @@ func (p *Player) Hurt(dmg float64, src world.DamageSource) (float64, bool) {
 			damageLeft -= a
 		} else {
 			p.SetAbsorption(a - damageLeft)
+
 			damageLeft = 0
 		}
 	}
+
+	if p.Health()-damageLeft <= mgl64.Epsilon {
+		hand, offHand := p.HeldItems()
+		if _, ok := hand.Item().(item.Totem); ok {
+			p.applyTotemEffects()
+			p.SetHeldItems(hand.Grow(-1), offHand)
+			return 0, false
+		} else if _, ok := offHand.Item().(item.Totem); ok {
+			p.applyTotemEffects()
+			p.SetHeldItems(hand, offHand.Grow(-1))
+			return 0, false
+		}
```

Comment:
Vanilla actually checks the offhand first for a totem, so we should probably match that behaviour too